azurerm/resource_arm_batch_pool.go

```diff
@@ -163,6 +163,46 @@ func resourceArmBatchPool() *schema.Resource {
 				Optional: true,
 				Default:  false,
 			},
+			"certificate": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:             schema.TypeString,
+							Required:         true,
+							ValidateFunc:     validate.NoEmptyStrings,
+							DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
Could we remove this>

Reply:
The DiffSuppressFunc on the certificate.id in the batch pool is needed because the Batch ARM API returns the ID with consistently difference casing between the Batch Certificate and the Batch Pool Certificate reference. So if you use Terraform references to set the Certificate ID in the Batch Pool from the Batch Certificate then you end up with a difference in case in the resulting Pool resource, which means that it looks like there are still changes to apply.

Reply:
The DiffSuppressFunc on the certificate.id in the batch pool is needed because the Batch ARM API returns the ID with consistently difference casing between the Batch Certificate and the Batch Pool Certificate reference. So if you use Terraform references to set the Certificate ID in the Batch Pool from the Batch Certificate then you end up with a difference in case in the resulting Pool resource, which means that it looks like there are still changes to apply.

Reply:
we should raise a bug about this API behaviour [on the Rest API Specs Repository](https://github.com/Azure/azure-rest-api-specs) since the Resource ID should be returned in a consistent manner (whilst the ARM spec _technically_ defines a Resource ID as case-insensitive; in practice both other ARM API's and downstream tooling don't handle these being case-insensitive - such that the casing for an ID should be the same within a Resource Provider API).

could we also add a TODO here with a link to the upstream API issue, when that's created?